src/2.md

```diff
@@ -89,27 +89,27 @@ The second example does not do this.
 
 Our string looks a little something like this (the left values are on the stack, the heap to the right):
 
-![](https://doc.rust-lang.org/book/img/trpl04-01.svg)
+{{#include ./img/trpl04-01.svg}}
 
 So to copy our data, we could copy both the heap and stack data, giving us a view in memory like this:
 
-![](https://doc.rust-lang.org/book/img/trpl04-03.svg)
+{{#include ./img/trpl04-03.svg}}
 
 This is also not what happens (imagine if your string was very very long, this would be a very expensive operation).
 
 So what we could do is copy only the data on the stack (the three numbers), giving us a view of the two variables that looks like this:
 
-![](https://doc.rust-lang.org/book/img/trpl04-02.svg)
+{{#include ./img/trpl04-02.svg}}
 
-This is not what happens. Remember that when variables go out of scope, they are `drop`ped and their memory is freed. If this were the case, the pointer would be freed twice, and this is a big error that can cause the heap to be corrupted.
+This is still dangerous. Remember that when variables go out of scope, they are `drop`ped and their memory is freed. If this were the case, the pointer would be freed twice, and this is a [big error](https://doc.rust-lang.org/nomicon/aliasing.html) that can cause the heap to be corrupted.
```

Comment:
not sure nomicon link is relevant, this restriction doesn't have much to do with aliasing, and pointer aliasing in rust is a whole other thing which is currently debated (strict provenance etc)